webui/src/data/server.js

```diff
@@ -285,11 +291,30 @@ class ServerCache {
 
             languages: null,
             disciplines: null,
+
+            currentCommunityRoles: {}, // a dictionary of current community's roles name and ID
```

Comment:
All the information needed for communities and roles can be retrieved by `this.props.params.id` and `serverCache.getCommunity(communityName)`! Don’t need to use `currentCommunityRoles` or `getCommunityRoles()` anymore.